cmd/thor/utils.go

```diff
@@ -437,98 +432,44 @@ func newP2PComm(ctx *cli.Context, repo *chain.Repository, txPool *txpool.TxPool,
 		return nil, errors.Wrap(err, "load or generate P2P key")
 	}
 
-	nat, err := nat.Parse(ctx.String(natFlag.Name))
+	userNAT, err := nat.Parse(ctx.String(natFlag.Name))
 	if err != nil {
 		cli.ShowAppHelp(ctx)
 		return nil, errors.Wrap(err, "parse -nat flag")
 	}
 
-	opts := &p2psrv.Options{
-		Name:            common.MakeName("thor", fullVersion()),
-		PrivateKey:      key,
-		MaxPeers:        ctx.Int(maxPeersFlag.Name),
-		ListenAddr:      fmt.Sprintf(":%v", ctx.Int(p2pPortFlag.Name)),
-		BootstrapNodes:  fallbackBootstrapNodes,
-		RemoteBootstrap: remoteBootstrapList,
-		NAT:             nat,
+	allowedPeers, err := parseNodeList(strings.TrimSpace(ctx.String(allowedPeersFlag.Name)))
```

Comment:
`if flagAllowedPeers != "" {` required for the input validation and since the empty list check was removed, then required outside

Reply:
An edge case for this might be --allowed-peers ",,," will give empty list, I think for this case explicitly declaring invalid input will make sense, wdyt?

Reply:
worth checking this case

Reply:
It's a good edge case, we're covered for it because of 
https://github.com/vechain/thor/pull/724/files#diff-cdfc7fee9b6dfccae42f1f5acc472fcaf660bfb5539d06c754a27502b12b8e00R590-R592

It will not try to parse a node because the string is "".